### PR TITLE
Fix observe object object instance

### PR DESCRIPTION
**Pull request by asummers**

### CODE REVIEW

File: leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/impl/ObjectResource.java [hunk 1]
```diff
@@ -297,13 +299,28 @@ public void removeObserverRelations() {
...
+                LwM2mPath observingPath = new LwM2mPath(relation.getExchange().getRequest().getOptions()
+                        .getUriPathString());
+                if (shouldNotify(observingPath, notifyingPath)) {
                     relation.notifyObservers();
                 }
             }
         }
     }
 
+    protected static boolean shouldNotify(LwM2mPath observingPath, LwM2mPath notifyingPath) {
```
Comment (louis89) [inline leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/impl/ObjectResource.java]: I would change this to "return observingPath.getObjectId() == notifyingPath.getObjectId() && (observingPath.getObjectInstanceId() == null || observingPath.getObjectInstanceId() == notifyingPath.getObjectInstanceId()) && (observingPath.getResourceId() == null || observingPath.getResourceId() == notifyingPath.getResourceId());"

File: leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/impl/ObjectResource.java [hunk 2]
```diff
@@ -297,13 +299,24 @@ public void removeObserverRelations() {
...
+        boolean sameObject = observingPath.getObjectId() == notifyingPath.getObjectId();
```
Comment (louis89) [inline leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/impl/ObjectResource.java]: oh dear god... Besides hating this style the names of these variables are misleading.

**Comment by jschloman** [comment]
Approve.

**Comment by louis89** [comment]
Approve

**Pull request merged**